pyjobshop/ProblemData.py

```diff
@@ -170,6 +170,41 @@ def name(self) -> Optional[str]:
         return self._name
 
 
+class OperationConstraint(StrEnum):
+    """
+    Constraint between two operations.
+
+    Constraints
+    -----------
+    - start_at_start:        $s(i) == s(j)$
+    - start_at_end:          $s(i) == f(j)$
+    - start_before_start:    $s(i) <= s(j)$
+    - start_before_end:      $s(i) <= f(j)$
+    - end_at_start:          $f(i) == s(j)$
+    - end_at_end:            $f(i) == f(j)$
+    - end_before_start:      $f(i) <= s(j)$
+    - end_before_end:        $f(i) <= f(j)$
+    - before:                i is right before j in sequence variable.
+    - previous:              i is previous to j in sequence variable.
+    - same_unit:             i and j are processed on the same unit.
+    - different_unit:        i and j are processed on different units.
+    """
+
+    START_AT_START = "start_at_start"
+    START_AT_END = "start_at_end"
+    START_BEFORE_START = "start_before_start"
+    START_BEFORE_END = "start_before_end"
+    END_AT_START = "end_at_start"
+    END_AT_END = "end_at_end"
+    END_BEFORE_START = "end_before_start"
+    END_BEFORE_END = "end_before_end"
+    PREVIOUS = "previous"
+    BEFORE = "before"
+    SAME_UNIT = "same_unit"
+    DIFFERENT_UNIT = "different_unit"
+    CONNECTED_UNITS = "connected_units"
+
```

Comment:
I find it hard to come up with a good name for these generalized precedence constraints, some of them which are not even precedence but simply relational.